Snakefile

```diff
@@ -131,11 +131,11 @@ config["conda_environment"] = CONDA_ENV_PATH
 
 # Define patterns we expect for wildcards.
 wildcard_constraints:
-    # Allow build names to contain alpha characters, underscores, and hyphens
+    # Allow build names to contain alphanumeric characters, underscores, and hyphens
     # but not special strings used for Nextstrain builds.
-    build_name = r'(?:[_a-zA-Z-](?!(tip-frequencies)))+',
+    build_name = r'(?:[a-zA-Z0-9_-](?!(tip-frequencies)))+',
     date = r"[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]",
-    origin = r"[a-zA-Z0-9-_]+"
+    origin = r"[a-zA-Z0-9_-]+"
```

Comment:
I had an opinion on hyphens at the end of regex sets here https://github.com/nextstrain/augur/pull/832#discussion_r793095816 but less of an opinion now, fine with this change.

Reply:
Oops, yes, I remember this now. I'm of the same mind as @tsibley about the convention of always placing hyphens at the end. Technically, we can always place them at the beginning instead and they will still get treated as literals but will be more "append-proof". Does that seem like a good compromise?

Reply:
Either way is fine to me. Though I don't think `-a-zA-Z0-9_` is much more "readable" than `a-zA-Z0-9-_`, so maybe it's best to have it at the end. I assume problems would be caught somewhere if we were to unintentionally append and make it a range like `_-~` in the future?

Reply:
Yeah, I would expect the readability to stem from the convention of first-or-last placement. I also don't expect us to update these too much, but it's good to have a convention within the team across projects that we're happy with. If you don't have a strong preference, I'll put the hyphen at the start to get the best of both worlds in terms of following convention and supporting appends to the end.